server/types/text.go

```diff
@@ -178,7 +179,10 @@ func (b TextType) SerializedCompare(v1 []byte, v2 []byte) (int, error) {
 	} else if len(v1) == 0 && len(v2) > 0 {
 		return -1, nil
 	}
-	return serializedStringCompare(v1, v2), nil
+
+	// Note that |v1| and |v2| are the addresses of the two TEXT values being compared, so instead of
+	// loading the large TEXT values into memory, we just compare their content-hashed addresses.
+	return bytes.Compare(v1, v2), nil
```

Comment:
If these are addresses, then I think something isn't working quite right. It's possible that there's a different code path that's calling this that's returning byte addresses, but this seems to work in other places that it's called (at least, it appears to be sorting properly without any `ORDER BY` directives). This requires further investigation.

Reply:
After digging into this more... these values seems to be addresses only when a `TEXT` field is used in a secondary index. In GMS, `ExtendedTupleComparator` (as well as `DefaultTupleComparator`) treat address encodings the same as non-address encodings, which works fine for Dolt, since an address encoded field won't ever be in an secondary index (not without a prefix length of content in-line at least). 

PostgreSQL doesn't have the same requirement on prefix lengths though, so I spent a little time digging into how much work that would be to support. It seems like TupleComparator would have to get some big changes to enable it to de-ref address-encoded values. In the end, I decided to just apply an implicit prefix length for Doltgres, since that's much smaller in scope. I'll close out this PR and open a new one for that change.